src/EPR.Calculator.API/Services/TransposePomAndOrgDataService.cs

```diff
@@ -130,5 +155,47 @@ public void Transpose(CalcResultsRequestDto resultsRequestDto)
                 }
             }
         }
+
+        private List<OrganisationDetails> GetOrganisationDetailsBySubmissionPeriod(int orgMasterId)
+        {
+            return [.. (from org in context.CalculatorRunOrganisationDataDetails
+                    join pom in context.CalculatorRunPomDataDetails
+                    on new { org.OrganisationId, org.SubmissionPeriodDesc } equals new { pom.OrganisationId, pom.SubmissionPeriodDesc }
+                    where (org.CalculatorRunOrganisationDataMasterId == orgMasterId && org.OrganisationName != null && org.SubsidaryId ==null)
```

Comment:
Should you add a condition to only get RunId specific values?

Reply:
RunId has a unique masterid